src/planner.c

```diff
@@ -422,6 +435,949 @@ timescaledb_set_rel_pathlist_query(PlannerInfo *root, RelOptInfo *rel, Index rti
+	 * of pages it will read and the number of tuples it will return.  (Note:
+	 * we assume the function returns sane values.)
+	 */
+	tsm = GetTsmRoutine(tsc->tsmhandler);
+	tsm->SampleScanGetSampleSize(root, rel, tsc->args,
+								 &pages, &tuples);
+
+	/*
+	 * For the moment, because we will only consider a SampleScan path for the
+	 * rel, it's okay to just overwrite the pages and tuples estimates for the
+	 * whole relation.  If we ever consider multiple path types for sampled
+	 * rels, we'll need more complication.
+	 */
+	rel->pages = pages;
+	rel->tuples = tuples;
+
+	/* Mark rel with estimated output rows, width, etc */
+	set_baserel_size_estimates(root, rel);
+}
+
+/* copied from allpaths.c */
+static void
+set_plain_rel_size(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte)
+{
+	/*
+	 * Test any partial indexes of rel for applicability.  We must do this
+	 * first since partial unique indexes can affect size estimates.
+	 */
+	check_index_predicates(root, rel);
+
+	/* Mark rel with estimated output rows, width, etc */
+	set_baserel_size_estimates(root, rel);
+}
+
+/* extracted from the same function in allpaths.c
+ * assumes that the root table is either excluded by constraints, or is an
+ * inheritance base table, and that chunks are regular tables
+ */
+static void
+set_rel_size(PlannerInfo *root, RelOptInfo *rel,
+			 Index rti, RangeTblEntry *rte)
+{
+	if (rel->reloptkind == RELOPT_BASEREL &&
+		relation_excluded_by_constraints(root, rel, rte))
+	{
+		/*
+		 * We proved we don't need to scan the rel via constraint exclusion,
+		 * so set up a single dummy path for it.  Here we only check this for
+		 * regular baserels; if it's an otherrel, CE was already checked in
+		 * set_append_rel_size().
+		 *
+		 * In this case, we go ahead and set up the relation's path right away
+		 * instead of leaving it for set_rel_pathlist to do.  This is because
+		 * we don't have a convention for marking a rel as dummy except by
+		 * assigning a dummy path to it.
+		 */
+		set_dummy_rel_pathlist(rel);
+	}
+	else if (rte->inh)
+	{
+		/* It's an "append relation", process accordingly */
+		set_append_rel_size(root, rel, rti, rte);
+	}
+	else
+	{
+		switch (rel->rtekind)
+		{
+			case RTE_RELATION:
+				if (rte->relkind == RELKIND_FOREIGN_TABLE)
+				{
+					/* Foreign table */
+					set_foreign_size(root, rel, rte);
+				}
+				else if (rte->relkind == RELKIND_PARTITIONED_TABLE)
+				{
+					/*
+					 * We could get here if asked to scan a partitioned table
+					 * with ONLY.  In that case we shouldn't scan any of the
+					 * partitions, so mark it as a dummy rel.
+					 */
+					set_dummy_rel_pathlist(rel);
+				}
+				else if (rte->tablesample != NULL)
+				{
+					/* Sampled relation */
+					set_tablesample_rel_size(root, rel, rte);
+				}
+				else
+				{
+					/* Plain relation */
+					set_plain_rel_size(root, rel, rte);
+				}
+				break;
+			case RTE_SUBQUERY:
```

Comment:
Was this modification necessary? I see that the code between 11.4 and 12.1 is the same or very similar here.

Reply:
It's not really necessary, it's just that these kinds should never appear as chunks. (the reason this fuction is copied in isn't because we need to make these modifications, they're just sanity-checking anyway, but because the original postgres function is marked `static`, and only called at a point where hypertables aren't expanded, providing the wrong info)